### PR TITLE
Add search feature

### PR DESCRIPTION
**Pull request by dmonzonis**
Now it is possible to filter records searching by name. The search
functionality can be accessed through a new icon on the toolbar, which
opens an edit text in which the user can type. The contents entered will
be used to filter records by name, ignoring case, and the recordset will
update as the user types.

Like other filters, the search query is persistent between tabs/games.

This commit also gets rid of the FAB completely, as it has no use for
now.

### CODE REVIEW

File: app/src/main/java/com/dmonzonis/nookpendium/FilterManager.kt [hunk 1]
```diff
@@ -21,6 +22,10 @@ class FilterManager(private val activity: CollectionActivity) {
     }
 
     fun applyFilters(recordset: Recordset) {
+        // Apply search filter separately
+        // TODO: Search by price if the searchQuery is a number
```
Comment (Elkasitu) [inline app/src/main/java/com/dmonzonis/nookpendium/FilterManager.kt:26]: idk about this, seems weird, what we could have instead is a filter with ranges for prices (from 2000 up, 2000 to 5000, etc)

File: app/src/main/java/com/dmonzonis/nookpendium/Record.kt [hunk 2]
```diff
@@ -28,10 +28,6 @@ class Recordset(private val totalRecords: List<Record>) {
         return records
     }
 
-    fun applyFilterByMonth(month: Int): List<Record> {
```
Comment (Elkasitu) [inline app/src/main/java/com/dmonzonis/nookpendium/Record.kt:31]: this is still possible with the new menu right?

**Review by Elkasitu — APPROVED** [review]

File: app/src/main/java/com/dmonzonis/nookpendium/Record.kt (hunk 2 above)
Comment (dmonzonis) [reply app/src/main/java/com/dmonzonis/nookpendium/Record.kt:31]: yes, it's just that the filter lambda is passed by the FilterManager, so Recordset just applies the filter to the records

File: app/src/main/java/com/dmonzonis/nookpendium/FilterManager.kt (hunk 1 above)
Comment (dmonzonis) [reply app/src/main/java/com/dmonzonis/nookpendium/FilterManager.kt:26]: I guess, it was just an idea

**Pull request merged**